pkg/tools/topo-generator/generator/writer.go

```diff
@@ -0,0 +1,67 @@
+package generator
+
+import (
+	"html/template"
+    "os"
+	//"topo-generator/parser"
+	"github.com/onosproject/onos-topo/pkg/tools/topo-generator/parser"
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+)
+
+var log = logging.GetLogger()
+
+func WriteFile(underlay parser.Underlay, filename string) {
+	// writing the entity-kind-relationship file
+	file, _ := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+
+	for _, n := range underlay.Networks {
+		// network
+		t, err := template.ParseFiles("github.com/onosproject/onos-topo/pkg/tools/topo-generator/generator/templates/network.yaml")
+		if err != nil {
+			log.Fatal(err.Error())
+			return
+		}
+		err = t.Execute(file, n)
+		
+		// switches
+		for _, s := range n.Switches {
+			t, err := template.ParseFiles("github.com/onosproject/onos-topo/pkg/tools/topo-generator/generator/templates/switch.yaml")
+			if err != nil {
```

Comment:
Use relative path rather than full path and define constants on top of the file.

Reply:
I tried using the relative path locally and it did not work (I had it at "templates/switch.yaml"), but that was not the same setup as it is here in onos-topo. Should I still change it to "templates/switch.yaml"?